src/compute-client/src/controller.rs

```diff
@@ -627,51 +622,42 @@ where
     }
 
     /// Processes the work queued by [`ComputeController::ready`].
-    ///
-    /// This method is guaranteed to return "quickly" unless doing so would
-    /// compromise the correctness of the system.
-    ///
-    /// This method is **not** guaranteed to be cancellation safe. It **must**
-    /// be awaited to completion.
-    pub async fn process(&mut self) -> Result<Option<ComputeControllerResponse<T>>, StorageError> {
+    pub fn process(&mut self) -> Option<ComputeControllerResponse<T>> {
```

Comment:
Its kind of surprising that this isn't `async` anymore. Doesn't this mean we could inline all the work and remove the process completely?

Reply:
We could. `ready` would be cancellation safe even if we did all the response handling there as well.

Two reasons I'd like to keep it this way:
* It's consistent with the storage controllers API (except that there the `process` method is still async).
* It is possible that `process` will become async again (i.e. when we introduce compute command durability). If we merged it into `ready`, there would be some risk that we will have forgotten about all the cancellation-safety woes again, and repeat our mistakes from the past. The presence of the two methods increases the chances that we won't forget.